tests/shell/tests.py

```diff
@@ -17,3 +19,13 @@ def test_command_option(self):
             )
             self.assertEqual(len(logger), 1)
             self.assertEqual(logger[0], __version__)
+
+    @mock.patch('django.core.management.commands.shell.select')
```

Comment:
I don't see how we could possibly pass an instance of OutputWrapper to `exec()`, so passing stdout to call_command has just no effect.

For stdin, I did the same [hack as in createsuperuser](https://github.com/django/django/blob/master/django/contrib/auth/management/commands/createsuperuser.py#L62) so it's more consistent. But if you want my own opinion, I would discourage adding production that's supposed to be used only in tests, I thought this was commonly known as a bad practice but am not sure anymore.